mysqltest/README.md

```diff
@@ -0,0 +1,91 @@
+# Mysqltest library
+
+## Introduction
+
+This is a Go library for starting an ephemeral MySQL Docker container. It is used to test
+integration between Go code and MySQL.
+
+## How to use it
+
+Since it takes about 15 second to start the MySQL Docker container, we recommend sharing a single
+MySQL instance between all of your tests. `USE` separate databases/namespaces or separate tables to
+isolate your tests from each other.
+
+Call `MustStart()` from your `TestMain()` function, If you're not familiar with Go's `TestMain()`
+mechanism for global test initialization, see the docs: https://pkg.go.dev/testing#hdr-Main.
+
+```
+import (
+    "database/sql"
+    "fmt"
+    "os"
+    "testing"
+
+    "github.com/abcxyz/pkg/mysqltest"
+    _ "github.com/go-sql-driver/mysql" // Link with the Go MySQL driver
+)
+
+var ci mysqltest.ConnInfo
+
+// TestMain runs once at startup to do test initialization common to all tests.
+func TestMain(m *testing.M) {
+    var closer io.Closer
+    ci, closer = mysqltest.MustStart() // Start the docker container. Can also pass options.
+    defer closer.Close()
```

Comment:
This defer won't run because it calls `os.Exit`

Reply:
Good catch, thanks. I redid this whole thing. I decided not to try to handle the case where tests panic, because the `testing` framework doesn't guarantee that our defers will run in that case (see the reference I left in the code comment for more info).